src/main/scala/wolfendale/nunjucks/TemplateParser.scala

```diff
@@ -51,16 +51,51 @@ object TemplateParser {
 
       import NoWhitespace._
       P(
-        (`if` ~ partial)
-          .map(TemplateNode.If.ConditionalContent.tupled) ~ (elif ~ partial)
-          .map(TemplateNode.If.ConditionalContent.tupled)
-          .rep ~ (`else` ~ partial).? ~ endIf)
+        (`if` ~ partial).map(TemplateNode.If.ConditionalContent.tupled)
+          ~ (elif ~ partial).map(TemplateNode.If.ConditionalContent.tupled).rep
+          ~ (`else` ~ partial).?
+          ~ endIf
+      )
         .map {
           case (x, xs, e) =>
             TemplateNode.If(x +: xs, e)
         }
     }
 
+    def switchTag = {
+      def switch = {
+        import SingleLineWhitespace._
+        P( openTag ~ "switch" ~/ Parser.expression ~ closeTag)
+      }
+
+      def `case` = {
+        import SingleLineWhitespace._
+        P( openTag ~ "case" ~/ Parser.expression ~ closeTag)
+      }
+
+      def `default` = {
+        import SingleLineWhitespace._
+        P( openTag ~ "default" ~ closeTag)
+      }
+
+      def endSwitch = {
+        import SingleLineWhitespace._
+        P( openTag ~ "endswitch" ~ closeTag)
+      }
+
+      import NoWhitespace._
+      P(
+        switch
+          ~ (`case` ~ partial).map(a => TemplateNode.Switch.ConditionalContent(Seq(a._1), a._2)).rep
+          ~ (`default` ~ partial).?
+          ~ endSwitch)
+        .map {
+          case (x, xs, e) =>
+            TemplateNode.Switch(x, xs, e)
+        }
```

Comment:
`.map(TemplateNode.Switch.tupled)`

Reply:
Can't make this work unfortunately, it's resolving to the object instead of the case class